lib/ci-helper.ts

```diff
@@ -150,6 +151,58 @@ export class CIHelper {
             console.timeEnd(`Making ${workDir} non-shallow`);
         };
         await unshallow(this.workDir);
+
+        if (setupOptions?.needsMailingListMirror) {
+            this.mailingListMirror = "mailing-list-mirror.git";
+            const epoch = this.config.mailrepo.public_inbox_epoch ?? 1;
+
+            // eslint-disable-next-line security/detect-non-literal-fs-filename
+            if (!fs.existsSync(this.mailingListMirror)) {
+                await git(["init", "--bare", "--initial-branch", this.config.mailrepo.branch, this.mailingListMirror]);
+            }
+
+            // First fetch from GitGitGadget's mirror, which supports partial clones
+            for (const [key, value] of [
+                ["remote.mirror.url", this.config.mailrepo.mirrorURL || this.config.mailrepo.url],
+                ["remote.mirror.promisor", "true"],
+                ["remote.mirror.partialCloneFilter", "blob:none"],
+            ]) {
+                await git(["config", key, value], { workDir: this.mailingListMirror });
+            }
+            console.time("fetch mailing list mirror");
+            await git(
+                [
+                    "-c",
+                    "remote.mirror.promisor=false", // let's fetch the mails with all of their contents
+                    "fetch",
+                    "mirror",
+                    "--depth=50",
+                    "+REF:REF".replace("REF", this.config.mailrepo.mirrorRef || `refs/heads/lore-${epoch}`),
```

Comment:
Does `mirrorRef` need to include `refs/heads/`?  I suppose you _could_ put things pretty much anywhere you want in a repo.